libpod/container_api.go

```diff
@@ -639,6 +639,32 @@ func (c *Container) Sync() error {
 	return nil
 }
 
+// ReloadNetwork reconfigures the container's network.
+// Technically speaking, it will tear down and then reconfigure the container's
+// network namespace, which will result in all firewall rules being recreated.
+// It is mostly intended to be used in cases where the system firewall has been
+// reloaded, and existing rules have been wiped out. It is expected that some
+// downtime will result, as the rules are destroyed as part of this process.
+// At present, this only works on root containers; it may be expanded to restart
+// slirp4netns in the future to work with rootless containers as well.
+// Requires that the container must be running or created.
+func (c *Container) ReloadNetwork() error {
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return err
+		}
+	}
+
+	if !c.ensureState(define.ContainerStateCreated, define.ContainerStateRunning) {
+		return errors.Wrapf(define.ErrCtrStateInvalid, "cannot reload network unless container network has been configured")
+	}
```

Comment:
@mheon Why did you accept `ContainerStateCreated`? Shouldn't we only do this for running containers?

Reply:
Created containers have had their network namespaces created already, so they also may need to have their network re-configured. Internally, we have ContainerStateConfigured (container exists in Libpod but has not been mounted or created in the OCI runtime) and ContainerStateCreated (container has been mounted, network has been configured, container is created in the runtime - but has not yet been started).

Reply:
OK, thanks for the clarification.